swan-lake/development-tutorials/test-document-the-code/ballerina-profiler.md

```diff
@@ -0,0 +1,365 @@
+---
+title: Ballerina Profiler
+description: Check out how you can run and do a CPU profile of a Ballerina program.
+keywords: ballerina runtime, troubleshoot, profiler, CPU profile, flame graph
+permalink: /learn/ballerina-profiler/
+active: ballerina-profiler
+---
+
+Improvements in your Ballerina code might be required to increase its runtime performance, minimize code execution times, increase throughput, and reduce latency. To do that, we need to identify performance bottlenecks, implications of concurrent executions, and find which code segments take longer to execute, and areas to improve performance. This can be done by using a profiler.
+
+Ballerina Profiler is a tool that monitors Ballerina runtime and its operations such as function calls. It can be used to understand the behavior and troubleshoot the performance issues of a Ballerina program and optimize it.
+
+>**Note:** Ballerina Profiler is an experimental feature, which supports only a limited set of functionality.
```

Comment:
```suggestion
>**Note:** Ballerina Profiler is an experimental feature, which supports only a limited set of functionality.
```
Should we mention that "and will be enhanced in future releases? Check the standard note we previously added for experimental features for persistence etc.)

Reply:
Checked https://ballerina.io/downloads/swan-lake-release-notes/swan-lake-2201.5.0#persist-tool. @warunalakshitha do we need to also say `The commands associated with the tool might change in future releases.`?